templates/ecs.yaml

```diff
@@ -0,0 +1,317 @@
+                  - !Sub 'arn:aws:kms:${AWS::Region}:${AWS::AccountId}:key/*'
+              - Effect: Allow
+                Action:
+                  - logs:CreateLogStream
+                Resource: '*'
+  ECSCluster:
+    Type: AWS::ECS::Cluster
+
+  # Security group for ECS nodes
+  # Should allow ephemeral ports from the ALB SG
+  # Should allow SSH from the Bastion hosts security group
+  EcsSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: ECS Node Security Group
+      VpcId: !Ref 'VpcId'
+  EcsSecurityGroupALBports:
+    Type: AWS::EC2::SecurityGroupIngress
+    Properties:
+      GroupId: !Ref 'EcsSecurityGroup'
+      IpProtocol: tcp
+      FromPort: 31000
+      ToPort: 61000
+      SourceSecurityGroupId: !Ref 'EcsAlbSecurityGroup'
+  EcsSecurityGroupSSHfromBastion:
+    Type: AWS::EC2::SecurityGroupIngress
+    Properties:
+      GroupId: !Ref 'EcsSecurityGroup'
+      IpProtocol: tcp
+      FromPort: 22
+      ToPort: 22
+      SourceSecurityGroupId: !Ref 'BastionHostSgId'
+
+  # Security group for the ALB that goes in front
+  # Should only allow port 80 from the world for now
+  EcsAlbSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: Load Balancer for ECS security group
+      VpcId: !Ref 'VpcId'
+  EcsAlbSecurityGroupHTTPinbound:
+    Type: AWS::EC2::SecurityGroupIngress
+    Properties:
+      GroupId: !Ref 'EcsAlbSecurityGroup'
+      IpProtocol: tcp
+      FromPort: 80
+      ToPort: 80
+      CidrIp: 0.0.0.0/0
+
+  CloudwatchLogsGroup:
+    Type: AWS::Logs::LogGroup
+    Properties:
+      LogGroupName: !Join ['-', [ECSLogGroup, !Ref 'AWS::StackName']]
+      RetentionInDays: 14
+
+  # The 'Family' property is key here, it tells ECS to increment the version rather
+  # than create a new Task Definition
+  taskdefinition:
+    Type: AWS::ECS::TaskDefinition
+    Properties:
+      Family: !Ref 'AWS::StackName'
+      ExecutionRoleArn: !GetAtt ['IAtlasIAMExecutionRole', 'Arn']
+      ContainerDefinitions:
+      - Name: iatlas-api
+        Cpu: 100
+        Essential: true
+        Image: !Join [':', ['registry.gitlab.com/cri-iatlas/iatlas-api', !Ref 'DockerImageTag']]
+        RepositoryCredentials:
+          CredentialsParameter: !Sub 'arn:aws:secretsmanager:${AWS::Region}:${AWS::AccountId}:secret:staging/gitlab-registry-credentials'
+        Memory: 300
+        LogConfiguration:
+          LogDriver: awslogs
+          Options:
+            awslogs-group: !Ref 'CloudwatchLogsGroup'
+            awslogs-region: !Ref 'AWS::Region'
+            awslogs-stream-prefix: iatlas-api
+        PortMappings:
+        - ContainerPort: 80
+        Environment:
+          - Name: FLASK_ENV
+            Value: 'production'
+        Secrets:
+          - Name: POSTGRES_DB
+            ValueFrom: !Sub 'arn:aws:secretsmanager:${AWS::Region}:${AWS::AccountId}:secret:iatlas_api_creds_encrypted:dbname::'
```

Comment:
these do not look like ARNs.  ARNs don't end in a `::` do they?

Also i don't see any of these parmeters in our AWS account's secretsmanager.

Reply:
This is the method recommended in AWS documentation for using a secret when you don't know (or need to specify) the version.

Reply:
@zaro0508 is this okay?

Reply:
references looks good.  although we will need to manually add these secrets to our aws account before deploying this.